### PR TITLE
Adding some capabilities

### PR DESCRIPTION
**Pull request by wzj1695224**
Adding actions, such as drawing a square and drawing from center.
Zoom in source image and zoom out capature image, you can shift their
scale factor.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,26 @@
 Image Clipper
 -------------
 
-This is a fork of the [imageclipper](https://code.google.com/p/imageclipper/)
+This is a fork of the [imageclipper](https://github.com/JoakimSoderberg/imageclipper)
```
Comment (JoakimSoderberg) [inline README.md]: Remove this text since I don't want that merged

File: src/Actions.hpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
+#pragma once
```
Comment (JoakimSoderberg) [inline src/Actions.hpp]: This is compiler specific so please change to:
```
#ifndef __ACTIONS_HPP__
#define __ACTIONS_HPP_
...

#endif // __ACTIONS_HPP_
```

File: src/Actions.hpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			status->resize_watershed = true;
+		else if (radius <= param->circle.width)
+			status->move_watershed = true;
+	}
+
+	// move reatangle
+	else if ((param->rect.x < x && x < param->rect.x + param->rect.width) && (param->rect.y < y && y < param->rect.y + param->rect.height))
+		status->move_rect = true;
+
+	// otherwise
```
Comment (JoakimSoderberg) [inline src/Actions.hpp]: A bit redundant comment :)

File: src/imageclipper.cpp [hunk 4]
```diff
@@ -43,835 +44,745 @@
...
+	//system("mode con lines=50 cols=80");
+
+	// initialization
+	CvCallbackParam init_param = {
+		"<S> Save <F> Forward <SPACE> s and f <B> Backward <ESC> Exit",			// main window's name
+		"Cropped",																// cropped window's name
+		NULL,																	// source image pointer
+		cvRect(0, 0, 0, 0),														// selected rect
+		cvRect(0, 0, 0, 0),														// watershed
+		0,																		// rotate
```
Comment (JoakimSoderberg) [inline src/imageclipper.cpp]: You seem to have a very wide screen, all these comments are way off screen on mine. Could you try to keep the   line length below 120 chars?

**Comment by JoakimSoderberg** [comment]
Thank you for these changes! I will need to do a closer look at this since the changes are quite huge. And verify it works on all OS's

File: src/Actions.hpp (hunk 2 above)
Comment (wzj1695224) [inline src/Actions.hpp]: Thank you for your advice.

File: src/imageclipper.cpp (hunk 4 above)
Comment (wzj1695224) [inline src/imageclipper.cpp]: I'm sorry about that, TAB doesn't work well!

File: README.md [hunk 5]
```diff
@@ -4,17 +4,21 @@ Image Clipper
...
-Since I used [CMake](http://www.cmake.org/) it was easy to make one build that
-works on all platforms.
+This fork was made initially for myself to improve its capabilities. 
+
 
 From the original description:
 
-> It is often required to crop images manually fast for computer vision researchers
-> to gather training and testing image sets, e.g., cropping faces in images.
+> It is often required to crop images manually fast for computer vision researchers to gather training and testing image sets, e.g., cropping faces in images.
```
Comment (JoakimSoderberg) [inline README.md]: Please remove all changes to the README for this Pull Request, they are not relevant.

File: src/Global.h [hunk 6]
```diff
@@ -0,0 +1,58 @@
+#pragma once
```
Comment (JoakimSoderberg) [inline src/Global.h]: Again replace this for all places, compiler specific

File: src/imageclipper.cpp [hunk 7]
```diff
@@ -43,835 +44,745 @@
...
+	cout << "  Supported Image Types" << endl;
+	cout << "      bmp|dib|jpeg|jpg|jpe|png|pbm|pgm|ppm|sr|ras|tiff|exr|jp2" << endl;
 }
 
 /**
-* cvSetMouseCallback function
+* Print Application Usage
 */
-void mouse_callback( int event, int x, int y, int flags, void* _param )
+void printUsage()
```
Comment (JoakimSoderberg) [inline src/imageclipper.cpp]: I'd prefer if you did not rearrange the functions like this for no reason. It simply makes a messier diff. Please move them back... If the reason is that you need to have a function declared before you use it in another function, then do that in a header.

I might want to backport some old patches, and changes like this makes doing that really hard

**Comment by JoakimSoderberg** [comment]
Thank you for the cleanup, I thought I had scared you away with my comments :)

**Comment by JoakimSoderberg** [comment]
I squashed all your commits and merged that... So if you want your master to be in sync run the following git command:

```bash
git checkout master

# Just if you don't have the upstream repo added as a remote already.
git remote add upstream git@github.com:JoakimSoderberg/imageclipper.git

git pull --rebase upstream master
```

**Pull request merged**